source/chapter3-debian.rst

```diff
@@ -5,3 +5,176 @@ Debian
+    WORKDIR /home/build
+
+Docker configuration for armhf
+------------------------------
+Create a Dockerfile for armhf (Dockerfile.armhf) with following content:
+::
+    FROM arm32v7/debian:buster
+
+    # Upgrade system and Yocto Proyect basic dependencie
+    RUN apt-get update && apt-get -y upgrade && apt-get -y install \
+        fai-server fai-setup-storage \
+        qemu-utils procps pigz kpartx \
+        u-boot-tools dosfstools gdisk
+
+    # Clean up APT when done.
+    RUN apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+    # Replace dash with bash
+    RUN rm /bin/sh && ln -s bash /bin/sh
+
+    # Make /home/build the working directory
+    RUN mkdir -p /home/build
+
+    # Download fai config
+    RUN cd /home/build && git clone https://git.linaro.org/ci/fai.git && cd fai && git checkout -b WORKING origin/master && cd /home/build
+
+    WORKDIR /home/build
+
+Script to generate debian content:
+---------------------------------
+Script content (build-fai-linaro-debian.sh):
+::
+    #!/bin/sh
+    LOCALPATH=$(pwd)
+    cd fai
+    sudo fai-diskimage -v --cspace $(pwd) \
+         --hostname linaro-ledge-debian \
+         -S 4G \
+         --class SAVECACHE,BUSTER,DEBIAN,LINARO,LEDGE,RAW \
+         "$LOCALPATH"/work.raw
+    # copy fai log
+    sudo cp /var/log/fai/linaro-ledge-debian/last/fai.log ../fai.log
+
+    # extract debian content from raw
+    LOOPDEV=$(sudo losetup -f -P --show "$LOCALPATH"/work.raw)
+    sudo mount $LOOPDEV /mnt/
+    cd /mnt/
+    sudo tar cJf "$LOCALPATH"/rootfs-linaro-buster-raw-unknown.ext4 .
+    cd -
+    sudo umount /mnt
+
+
+Create docker image:
+-------------------
+For arm64
+
+.. code-block:: bash
+
+    docker build --force-rm --file Dockerfile.arm64 --tag creation .
+For armhf
+
+.. code-block:: bash
+
+    docker build --force-rm --file Dockerfile.armhf --tag creation .
+
+Execute docker image on interactive mode with priviledge:
+--------------------------------------------------------
+
+.. code-block:: bash
+
+    docker run -it --privileged -v $PWD:/home/build creation
+
+The current directory are shared with docker.
+
+
+
+Create Debian image:
+--------------------
+On Docker environment and fai directory
+
+.. code-block:: bash
+
+    DOCKER> ./build-fai-linaro-debian.sh
+    DOCKER> exit
+
+Image generation per board
+==========================
+For qemu (armhf):
+-----------------
+
+.. code-block:: bash
+
+    mkdir -p script
+    wget https://raw.githubusercontent.com/Linaro/meta-ledge/zeus/meta-ledge-bsp/recipes-devtools/generate-raw-image/raw-tools/create_raw_from_flashlayout.sh \
+     -O script/create_raw_from_flashlayout.sh
+    chmod +x script/create_raw_from_flashlayout.sh
+
+    wget https://raw.githubusercontent.com/Linaro/meta-ledge/zeus/meta-ledge-bsp/recipes-devtools/generate-raw-image/files/ledge-stm32mp157c-dk2/FlashLayout_sdcard_ledge-stm32mp157c-dk2-debian.tsv.template -O  FlashLayout_sdcard_ledge-stm32mp157c-dk2-debian.tsv.template
```

Comment:
where FlashLayout_sdcard_ledge-stm32mp157c-dk2-debian.tsv.template is used?

Reply:
Corrected